haystack/nodes/prompt/invocation_layer/hugging_face.py

```diff
@@ -168,19 +170,28 @@ def _prepare_pipeline_kwargs(self, **kwargs) -> Dict[str, Any]:
         torch_dtype = self._extract_torch_dtype(**kwargs)
         # and the model (prefer model instance over model_name_or_path str identifier)
         model = kwargs.get("model") or kwargs.get("model_name_or_path")
+        trust_remote_code = kwargs.get("trust_remote_code", False)
+
+        tokenizer = kwargs.get("tokenizer", None)
+
+        if isinstance(tokenizer, str):
+            model_config = AutoConfig.from_pretrained(model, trust_remote_code=trust_remote_code)
+            load_tokenizer = type(model_config) in TOKENIZER_MAPPING or model_config.tokenizer_class is not None
+            if not load_tokenizer:
+                tokenizer = AutoTokenizer.from_pretrained(model, trust_remote_code=trust_remote_code)
```

Comment:
Let's rename `load_tokenizer`, I find it quite confusing that we load a tokenizer in cases where `load_tokenizer` is `False`. Also, lets's add a comment explaining why this is necessary.

Reply:
Let's move this code to a separate method.

Reply:
sure! I am on it.